core/src/main/java/io/temporal/samples/hello/HelloUpdate.java

```diff
@@ -0,0 +1,281 @@
+ * config to enabled update.
+ */
+public class HelloUpdate {
+
+  // Define the task queue name
+  static final String TASK_QUEUE = "HelloUpdateTaskQueue";
+
+  // Define the workflow unique id
+  static final String WORKFLOW_ID = "HelloUpdateWorkflow";
+
+  /**
+   * The Workflow Definition's Interface must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow Definitions should not contain any heavyweight computations, non-deterministic
+   * code, network calls, database operations, etc. Those things should be handled by the
+   * Activities.
+   *
+   * @see WorkflowInterface
+   * @see WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface GreetingWorkflow {
+    /**
+     * This is the method that is executed when the Workflow Execution is started. The Workflow
+     * Execution completes when this method finishes execution.
+     */
+    @WorkflowMethod
+    List<String> getGreetings();
+
+    /*
+     * Define the workflow waitForName update method. This method is executed when the workflow
+     * receives an update request.
+     */
+    @UpdateMethod()
+    Integer waitForName(String name);
+
+    /*
+     * Define an optional workflow update validator. The validator must take the same parameters as the update handle.
+     * The validator is run before the update handle.
+     * If the validator fails by throwing any exception the update request will be rejected and the handle will not run.
+     */
+    @UpdateValidatorMethod(updateName = "waitForName")
+    void waitForNameValidator(String name);
+
+    // Define the workflow exit signal method. This method is executed when the workflow receives a
+    // signal.
+    @SignalMethod
+    void exit();
+  }
+
+  // Define the workflow implementation which implements the getGreetings workflow method.
+  public static class GreetingWorkflowImpl implements GreetingWorkflow {
+
+    // messageQueue holds up to 10 messages (received from updates)
+    List<String> messageQueue = new ArrayList<>(10);
+    List<String> receivedMessages = new ArrayList<>(10);
+    boolean exit = false;
+
+    private final HelloActivity.GreetingActivities activities =
+        Workflow.newActivityStub(
+            HelloActivity.GreetingActivities.class,
+            ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(2)).build());
+
+    @Override
+    public List<String> getGreetings() {
+
+      while (true) {
+        // Block current thread until the unblocking condition is evaluated to true
+        Workflow.await(() -> !messageQueue.isEmpty() || exit);
+        if (messageQueue.isEmpty() && exit) {
+          // no messages in queue and exit signal was sent, return the received messages
+          return receivedMessages;
+        }
+        String message = messageQueue.remove(0);
+        receivedMessages.add(message);
+      }
+    }
+
+    @Override
+    public Integer waitForName(String name) {
```

Comment:
What does `waitForName` mean? Not sure "wait" is the obvious term for what is happening. Maybe just `addGreeting`?